pkg/router/functionHandler.go

```diff
@@ -289,8 +292,17 @@ func (roundTripper RetryingRoundTripper) RoundTrip(req *http.Request) (*http.Res
 
 		roundTripper.logger.Debug("request headers", zap.Any("headers", req.Header))
 
+		// Creating context for client
+		if roundTripper.timeout <= 0 {
+			roundTripper.timeout = fv1.DEFAULT_FUNCTION_TIMEOUT
+		}
+		roundTripper.logger.Debug("Creating context for request for ", zap.Any("Time", roundTripper.timeout))
+		var closeCtx func()
+		ctx, closeCtx := context.WithTimeout(context.Background(), time.Duration(roundTripper.timeout)*time.Second)
+		defer closeCtx()
```

Comment:
Actually, it's not necessary to declare `closeCtx` here, you can remove it.
Also, a possible resource leak problem: `defer in for loop`.[[reference](https://stackoverflow.com/questions/45617758/defer-in-the-loop-what-will-be-better)]

How about we changing this into
```
// forward the request to the function service
ctx, closeCtx := context.WithTimeout(context.Background(), time.Duration(roundTripper.timeout)*time.Second)
resp, err = ocRoundTripper.RoundTrip(req.WithContext(ctx))
closeCtx()
```

Reply:
Yes, you are right defer can cause a resource leak. Initially it was added since we were processing a request in a goroutine. I will correct it.